diff -up python3-nightly/Python/pylifecycle.c.old python3-nightly/Python/pylifecycle.c
--- python3-nightly/Python/pylifecycle.c.old	2015-02-07 13:38:56.279413689 +0100
+++ python3-nightly/Python/pylifecycle.c	2015-02-07 13:39:36.920664202 +0100
@@ -612,7 +612,15 @@ Py_Finalize(void)
 
     /* Debugging stuff */
 #ifdef COUNT_ALLOCS
-    dump_counts(stdout);
+    /* This is a downstream Fedora modification.
+       The upstream default with COUNT_ALLOCS is to always dump the counts to
+       stdout on exit.  For our debug builds its useful to have the info from
+       COUNT_ALLOCS available, but the stdout info here gets in the way, so
+       we make it optional, wrapping it in an environment variable (modelled
+       on the other PYTHONDUMP* env variables):
+    */
+    if (Py_GETENV("PYTHONDUMPCOUNTS"))
+        dump_counts(stdout);
 #endif
     /* dump hash stats */
     _PyHash_Fini();
