diff --git a/Lib/distutils/command/install.py b/Lib/distutils/command/install.py
index ed44a93..1af2240 100644
--- a/Lib/distutils/command/install.py
+++ b/Lib/distutils/command/install.py
@@ -36,9 +36,9 @@ INSTALL_SCHEMES = {
         'data'   : '$base',
         },
     'unix_home': {
-        'purelib': '$base/lib/python',
-        'platlib': '$base/lib64/python',
-        'headers': '$base/include/python/$dist_name',
+        'purelib': '$base/lib/platform-python',
+        'platlib': '$base/lib64/platform-python',
+        'headers': '$base/include/platform-python/$dist_name',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index 240bb66..57cebae 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -32,12 +32,12 @@ _INSTALL_SCHEMES = {
         'data': '{base}',
         },
     'posix_home': {
-        'stdlib': '{installed_base}/lib/python',
-        'platstdlib': '{base}/lib/python',
-        'purelib': '{base}/lib/python',
-        'platlib': '{base}/lib/python',
-        'include': '{installed_base}/include/python',
-        'platinclude': '{installed_base}/include/python',
+        'stdlib': '{installed_base}/lib/platform-python',
+        'platstdlib': '{base}/lib/platform-python',
+        'purelib': '{base}/lib/platform-python',
+        'platlib': '{base}/lib/platform-python',
+        'include': '{installed_base}/include/platform-python',
+        'platinclude': '{installed_base}/include/platform-python',
         'scripts': '{base}/bin',
         'data': '{base}',
         },
diff --git a/Makefile.pre.in b/Makefile.pre.in
index 4f8c042..cd30d06 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1449,7 +1449,7 @@ frameworkinstallstructure:	$(LDLIBRARY)
 		else	true; \
 		fi; \
 	done
-	$(LN) -fsn include/python$(LDVERSION) $(DESTDIR)$(prefix)/Headers
+	$(LN) -fsn include/platform-python$(LDVERSION) $(DESTDIR)$(prefix)/Headers
 	sed 's/%VERSION%/'"`$(RUNSHARED) ./$(BUILDPYTHON) -c 'import platform; print(platform.python_version())'`"'/g' < $(RESSRCDIR)/Info.plist > $(DESTDIR)$(prefix)/Resources/Info.plist
 	$(LN) -fsn $(VERSION) $(DESTDIR)$(PYTHONFRAMEWORKINSTALLDIR)/Versions/Current
 	$(LN) -fsn Versions/Current/$(PYTHONFRAMEWORK) $(DESTDIR)$(PYTHONFRAMEWORKINSTALLDIR)/$(PYTHONFRAMEWORK)
diff --git a/Misc/python-config.in b/Misc/python-config.in
index e13da75..a72893c 100644
--- a/Misc/python-config.in
+++ b/Misc/python-config.in
@@ -47,7 +47,7 @@ for opt in opt_flags:
         print(' '.join(flags))
 
     elif opt in ('--libs', '--ldflags'):
-        libs = ['-lpython' + pyver + sys.abiflags]
+        libs = ['-lplatform-python' + pyver + sys.abiflags]
         libs += getvar('LIBS').split()
         libs += getvar('SYSLIBS').split()
         # add the prefix/lib/pythonX.Y/config dir, but only if there is no
diff --git a/Misc/python-config.sh.in b/Misc/python-config.sh.in
index 30c6927..d62425b 100644
--- a/Misc/python-config.sh.in
+++ b/Misc/python-config.sh.in
@@ -40,19 +40,19 @@ LIBM="@LIBM@"
 LIBC="@LIBC@"
 SYSLIBS="$LIBM $LIBC"
 ABIFLAGS="@ABIFLAGS@"
-LIBS="-lpython${VERSION}${ABIFLAGS} @LIBS@ $SYSLIBS"
+LIBS="-lplatform-python${VERSION}${ABIFLAGS} @LIBS@ $SYSLIBS"
 BASECFLAGS="@BASECFLAGS@"
 LDLIBRARY="@LDLIBRARY@"
 LINKFORSHARED="@LINKFORSHARED@"
 OPT="@OPT@"
 PY_ENABLE_SHARED="@PY_ENABLE_SHARED@"
 LDVERSION="@LDVERSION@"
-LIBDEST=${prefix}/lib/python${VERSION}
+LIBDEST=${prefix}/lib/platform-python${VERSION}
 LIBPL=$(echo "@LIBPL@" | sed "s#$prefix_build#$prefix_real#")
 SO="@EXT_SUFFIX@"
 PYTHONFRAMEWORK="@PYTHONFRAMEWORK@"
-INCDIR="-I$includedir/python${VERSION}${ABIFLAGS}"
-PLATINCDIR="-I$includedir/python${VERSION}${ABIFLAGS}"
+INCDIR="-I$includedir/platform-python${VERSION}${ABIFLAGS}"
+PLATINCDIR="-I$includedir/platform-python${VERSION}${ABIFLAGS}"
 
 # Scan for --help or unknown argument.
 for ARG in $*
diff --git a/Misc/python.pc.in b/Misc/python.pc.in
index ae69867..38405a1 100644
--- a/Misc/python.pc.in
+++ b/Misc/python.pc.in
@@ -9,5 +9,5 @@ Description: Python library
 Requires: 
 Version: @VERSION@
 Libs.private: @LIBS@
-Libs: -L${libdir} -lpython@VERSION@@ABIFLAGS@
-Cflags: -I${includedir}/python@VERSION@@ABIFLAGS@
+Libs: -L${libdir} -lplatform-python@VERSION@@ABIFLAGS@
+Cflags: -I${includedir}/platform-python@VERSION@@ABIFLAGS@
diff --git a/Modules/makesetup b/Modules/makesetup
index 8db8de8..430e323 100755
--- a/Modules/makesetup
+++ b/Modules/makesetup
@@ -92,7 +92,7 @@ CYGWIN*) if test $libdir = .
 	 else
 	 	ExtraLibDir='$(LIBPL)'
 	 fi
-	 ExtraLibs="-L$ExtraLibDir -lpython\$(VERSION)";;
+	 ExtraLibs="-L$ExtraLibDir -lplatform-python\$(VERSION)";;
 esac
 
 # Main loop
diff --git a/configure b/configure
index eea17a4..0cdf419 100755
--- a/configure
+++ b/configure
@@ -5962,7 +5962,7 @@ $as_echo "#define Py_ENABLE_SHARED 1" >>confdefs.h
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lplatform-python$(LDVERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
@@ -5989,12 +5989,12 @@ $as_echo "#define Py_ENABLE_SHARED 1" >>confdefs.h
 			LDLIBRARY='libpython$(LDVERSION).sl'
 			;;
 	  esac
-	  BLDLIBRARY='-Wl,+b,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBRARY='-Wl,+b,$(LIBDIR) -L. -lplatform-python$(LDVERSION)'
 	  RUNSHARED=SHLIB_PATH=`pwd`${SHLIB_PATH:+:${SHLIB_PATH}}
 	  ;;
     Darwin*)
     	LDLIBRARY='libpython$(LDVERSION).dylib'
-	BLDLIBRARY='-L. -lpython$(LDVERSION)'
+	BLDLIBRARY='-L. -lplatform-python$(LDVERSION)'
 	RUNSHARED=DYLD_LIBRARY_PATH=`pwd`${DYLD_LIBRARY_PATH:+:${DYLD_LIBRARY_PATH}}
 	;;
     AIX*)
@@ -15060,9 +15060,9 @@ $as_echo "$LDVERSION" >&6; }
 
 
 if test x$PLATFORM_TRIPLET = x; then
-  LIBPL='$(prefix)'"/lib/python${VERSION}/config-${LDVERSION}"
+  LIBPL='$(prefix)'"/lib/platform-python${VERSION}/config-${LDVERSION}"
 else
-  LIBPL='$(prefix)'"/lib/python${VERSION}/config-${LDVERSION}-${PLATFORM_TRIPLET}"
+  LIBPL='$(prefix)'"/lib/platform-python${VERSION}/config-${LDVERSION}-${PLATFORM_TRIPLET}"
 fi
 
 
diff --git a/configure.ac b/configure.ac
index 74b0e57..f20b54e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1139,7 +1139,7 @@ if test $enable_shared = "yes"; then
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lplatform-python$(LDVERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
@@ -1148,13 +1148,13 @@ if test $enable_shared = "yes"; then
 	  fi
           ;;
     Linux*|GNU*|NetBSD*|FreeBSD*|DragonFly*|OpenBSD*)
-	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
+	  LDLIBRARY='libplatform-python$(LDVERSION).so'
+	  BLDLIBRARY='-L. -lplatform-python$(LDVERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
           then
-	      PY3LIBRARY=libpython3.so
+	      PY3LIBRARY=libplatform-python3.so
 	  fi
 	  ;;
     hp*|HP*)
@@ -1166,12 +1166,12 @@ if test $enable_shared = "yes"; then
 			LDLIBRARY='libpython$(LDVERSION).sl'
 			;;
 	  esac
-	  BLDLIBRARY='-Wl,+b,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBRARY='-Wl,+b,$(LIBDIR) -L. -lplatform-python$(LDVERSION)'
 	  RUNSHARED=SHLIB_PATH=`pwd`${SHLIB_PATH:+:${SHLIB_PATH}}
 	  ;;
     Darwin*)
-    	LDLIBRARY='libpython$(LDVERSION).dylib'
-	BLDLIBRARY='-L. -lpython$(LDVERSION)'
+		LDLIBRARY='libplatform-ython$(LDVERSION).dylib'
+	BLDLIBRARY='-L. -lplatform-python$(LDVERSION)'
 	RUNSHARED=DYLD_LIBRARY_PATH=`pwd`${DYLD_LIBRARY_PATH:+:${DYLD_LIBRARY_PATH}}
 	;;
     AIX*)
@@ -4691,9 +4691,9 @@ AC_MSG_RESULT($LDVERSION)
 dnl define LIBPL after ABIFLAGS and LDVERSION is defined.
 AC_SUBST(PY_ENABLE_SHARED)
 if test x$PLATFORM_TRIPLET = x; then
-  LIBPL='$(prefix)'"/lib/python${VERSION}/config-${LDVERSION}"
+  LIBPL='$(prefix)'"/lib/platform-python${VERSION}/config-${LDVERSION}"
 else
-  LIBPL='$(prefix)'"/lib/python${VERSION}/config-${LDVERSION}-${PLATFORM_TRIPLET}"
+  LIBPL='$(prefix)'"/lib/platform-python${VERSION}/config-${LDVERSION}-${PLATFORM_TRIPLET}"
 fi
 AC_SUBST(LIBPL)
 
